hooks/src/index.js

```diff
@@ -395,7 +379,29 @@ function hash(s) {
 export function useId() {
 	const state = getHookState(currentIndex++, 11);
 	if (!state._value) {
-		state._value = 'P' + hash(currentComponent._vnode._mask) + currentIndex;
+		// Traverse the tree upwards and count the components until we reach
+		// the root node.
+		let root = currentComponent._vnode;
+		let parent = root._parent;
+		let i = 0;
+		while (parent !== null) {
+			if (parent.type !== Fragment && typeof parent.type === 'function') {
+				i++;
+			}
+
+			root = parent;
+			parent = parent._parent;
+		}
+
+		// Attach the id usage array to the root node and resize it to fit
+		let ids = root._ids || (root._ids = [0]);
```

Comment:
That's not needed because we always wrap the passed vnode with a `Fragment` root node. Every new render to string call has its own unique root node, so we'll always create the new ids array.

Reply:
Nice!